var/spack/repos/builtin/packages/py-mkdocs/package.py

```diff
@@ -14,17 +14,28 @@ class PyMkdocs(PythonPackage):
     homepage = "https://www.mkdocs.org/"
     pypi = "mkdocs/mkdocs-1.3.1.tar.gz"
 
+    version("1.4.2", sha256="8947af423a6d0facf41ea1195b8e1e8c85ad94ac95ae307fe11232e0424b11c5")
     version("1.3.1", sha256="a41a2ff25ce3bbacc953f9844ba07d106233cd76c88bac1f59cb1564ac0d87ed")
 
     depends_on("python@3.6:", type=("build", "run"))
-    depends_on("py-setuptools", type="build")
+    depends_on("python@3.7:", when="@1.4.2:", type=("build", "run"))
+    depends_on("py-setuptools", when="@:1.3", type="build")
+    depends_on("py-hatchling", when="@1.4.2:", type="build")
     depends_on("py-click@3.3:", type=("build", "run"))
+    depends_on("py-click@7.0:", when="@1.4.2:", type=("build", "run"))
     depends_on("py-jinja2@2.10.2:", type=("build", "run"))
-    depends_on("py-markdown@3.2.1:3.3", type=("build", "run"))
+    depends_on("py-jinja2@2.11.1:", when="@1.4.2:", type=("build", "run"))
+    depends_on("py-markdown@3.2.1:3.3", when="@:1.3", type=("build", "run"))
+    depends_on("py-markdown@3.2.1:3.4", when="@1.4.2:", type=("build", "run"))
```

Comment:
Latest version is `<3.4` which is equivalent to `:3.3` in Spack where version bounds are inclusive

Reply:
So should I take out the upper bound entirely for line 28?

Reply:
You should change:
```python
    depends_on("py-markdown@3.2.1:3.3", when="@:1.3", type=("build", "run"))
    depends_on("py-markdown@3.2.1:3.4", when="@1.4.2:", type=("build", "run"))
```
to:
```python
    depends_on("py-markdown@3.2.1:3.3", type=("build", "run"))
```
since nothing has changed since the last release.